src/main/java/org/quiltmc/loader/impl/gui/QuiltGuiEntry.java

```diff
@@ -109,4 +161,74 @@ private static void addThrowable(QuiltStatusNode node, Throwable e, Set<Throwabl
 			addThrowable(sub, t, seen);
 		}
 	}
+
+	private static void fork(QuiltStatusTree tree, boolean shouldWait) throws Exception {
+		List<String> commands = new ArrayList<>();
+		commands.add(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java");
+		commands.add("-cp");
+		// Is this a good idea?
+		// I don't think we actually care about most of the classes here
+		// just *this* jar file?
+		commands.add(System.getProperty("java.class.path"));
+		commands.add(QuiltGuiEntry.class.getName());
+
+		File jsonFile = File.createTempFile("quilt-loader-tree", ".json.gz");
+
+		try (OutputStream out = new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(jsonFile)))) {
+			try (JsonWriter writer = JsonWriter.create(new OutputStreamWriter(out, StandardCharsets.UTF_8))) {
+				writer.setIndent(" ");
+				tree.write(writer);
+				writer.flush();
+			}
+		}
+		commands.add("--read-tree");
+		commands.add(jsonFile.getAbsolutePath());
+
+		ProcessBuilder pb = new ProcessBuilder(commands);
+		pb.redirectError(Redirect.PIPE);
+		// TODO: handle more complex button presses by sending data over sysout?
+		pb.redirectOutput(Redirect.PIPE);
+
+		Process p = pb.start();
+
+		// TODO: Handle input (like "Status: Correct tree", or "Status: Continue")
+
+		jsonFile.deleteOnExit();
+
+		if (!shouldWait) {
+			return;
+		}
+
+		int result = p.waitFor();
+
+		if (result != 0) {
+			throw new Exception("Failed to open the gui! (The error should be higher up in the log/output)");
+		}
+	}
+
+	/** The entry point after forking the main application over into a different process to get around incompatibilities
+	 * on OSX (and problems where some games switch swing over to use headless mode, which doesn't work very well for
+	 * us). */
+	public static void main(String[] args) throws Exception {
+		if (args.length == 2 && "--read-tree".equals(args[0])) {
+			System.out.println("Status:Reading");
+			File from = new File(args[1]);
+
+			try (InputStream is = new GZIPInputStream(new BufferedInputStream(new FileInputStream(from)))) {
+				JsonReader reader = JsonReader.create(new InputStreamReader(is, StandardCharsets.UTF_8));
```

Comment:
is creating a json5 reader intentional here? I probably need to change these names one more time...

Reply:
Am I meant to use `new JsonReader(Reader)` instead? For some reason I don't have docs for json5 in my IDE.

Or am I not meant to be using `quilt-json5` for this at all? I'm currently assuming we need to keep loader far away from any dependencies, which I assumed we want to use json5 for json handling?

(Either way, what I'm trying to do here is read and write a json file automatically, that the user isn't expected to see normally. For the sake of debugging it's exported with a single space for indenting, but there's definitely no need for comments).

Reply:
Use .createStrict here

Reply:
I think we should flip the options glitch, this is a confusion I see quite often

Reply:
ok

Reply:
See the issue I opened on the github for it to pitch in name change ideas. i know it needs to change before it's stabilized but idk to what